plastic_tui/src/ui.rs

```diff
@@ -451,14 +451,19 @@ impl Ui {
 
     fn handle_gamepad(&mut self) {
         // set events in the cache and check if gamepad is still active
-        while let Some(GilrsEvent { id, event, .. }) = self.gilrs.next_event() {
+        if self.gilrs.is_none() {
```

Comment:
A small improvement here,
you can use 
```rust
// didn't test the code, but the idea here
let Some(mut gilrs) = self.gilrs else {
    return;
}
```
then afterwards, you can use `girls` without the need for `as_mut().unwrap()`

Reply:
Thanks for the suggestion, will implement

Reply:
You can use ref